core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java

```diff
@@ -221,9 +221,15 @@ public void free(AlluxioURI path, final FreePOptions options)
   @Override
   public List<BlockLocationInfo> getBlockLocations(AlluxioURI path)
       throws IOException, AlluxioException {
-    List<BlockLocationInfo> blockLocations = new ArrayList<>();
     // Don't need to checkUri here because we call other client operations
-    List<FileBlockInfo> blocks = getStatus(path).getFileBlockInfos();
+    return getBlockLocations(getStatus(path), path);
+  }
+
+  @Override
+  public List<BlockLocationInfo> getBlockLocations(URIStatus status, AlluxioURI path)
```

Comment:
I think you can get the path from `status.getPath()`, so why do you need to add a new parameter. @LuQQiu

Reply:
Add a new method to let getStatus can go through metadataCachingFileSystem and utilized the cached metedata

Reply:
I mean that you don't need the parameter `AlluxioURI path`.

Reply:
good call!
Yeah it's not needed, it may help to reduce the string <--> AlluxioURI type change and new object construction, but from caller perspective, the method should not require both information since one contains the other